### PR TITLE
feat: update to React router 6

### PR DESCRIPTION
**Pull request by cstenglein**
Worked on that before Christmas last time, but better to make a draft than let it just sit around

TODOS:

- [x] Redirect to /login correctly if not logged in (currently throws a state update error)
- [x] Save login path if not logged in and redirect to requested page after successful login

### CODE REVIEW

File: src/components/Navigation/Header/DropdownMenu/DropdownMenu.tsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-import { ReactComponent as LogoutIcon } from "../../../../assets/logout.svg";
 
 const DropdownMenu = forwardRef<HTMLDivElement>((_, ref) => {
   const { t } = useTranslation();
-  const appCtx = useContext(AppContext);
-  const unitActive = appCtx.unit === "Sat";
-  const darkActive = appCtx.darkMode;
+  const { unit, darkMode, logout, toggleDarkMode, toggleUnit } =
+    useContext(AppContext);
+  const unitActive = unit === "Sat";
```
Comment (escapedcat) [inline src/components/Navigation/Header/DropdownMenu/DropdownMenu.tsx:11]: Not related to this MR but could we make this an enum?

File: src/components/Navigation/SideDrawer/SideDrawer.tsx [hunk 2]
```diff
@@ -24,28 +24,31 @@ export const SideDrawer: FC = () => {
       <div>
         <NavLink
           to="/home"
-          className={navLinkClasses}
-          activeClassName={navLinkActiveClasses}
+          className={({ isActive }) =>
+            navLinkClasses + (isActive ? " " + navLinkActiveClasses : "")
```
Comment (escapedcat) [inline src/components/Navigation/SideDrawer/SideDrawer.tsx:28]: What about i.e. this?: 

```js
`${navLinkClasses} ${isActive ? navLinkActiveClasses : ''}`
```

Comment (cstenglein) [reply src/components/Navigation/SideDrawer/SideDrawer.tsx:28]: could do that some other time & also move out the navlink classes, they don't need to be created on every render.

Comment (escapedcat) [reply src/components/Navigation/SideDrawer/SideDrawer.tsx:28]: 👍  wannna create an issue for that?

**Review by escapedcat — APPROVED** [review]

File: src/components/Navigation/SideDrawer/SideDrawer.tsx (hunk 2 above)
Comment (cstenglein) [reply src/components/Navigation/SideDrawer/SideDrawer.tsx:28]: If you want you can :) 
I don't think it's that urgent atm.

**Pull request merged**